testutils/src/main/scala/za/co/absa/enceladus/testutils/infoFileComparison/AtumModelUtils.scala

```diff
@@ -0,0 +1,192 @@
+      */
+    def compareWith(secondMetadata: ControlMeasureMetadata, curPath: String): List[ModelDifference[_]] = {
+      val diffs = simpleCompare(metadata.sourceApplication,
+        secondMetadata.sourceApplication, s"$curPath.sourceApplication") ::
+      simpleCompare(metadata.country, secondMetadata.country, s"$curPath.country") ::
+      simpleCompare(metadata.historyType, secondMetadata.historyType, s"$curPath.historyType") ::
+      simpleCompare(metadata.dataFilename, secondMetadata.dataFilename, s"$curPath.dataFilename") ::
+      simpleCompare(metadata.sourceType, secondMetadata.sourceType, s"$curPath.sourceType") ::
+      simpleCompare(metadata.version, secondMetadata.version, s"$curPath.version") ::
+      simpleCompare(metadata.informationDate, secondMetadata.informationDate, s"$curPath.informationDate") ::
+        Nil
+
+      val additionalInfoDiff = additionalInfoComparison(metadata.additionalInfo,
+        secondMetadata.additionalInfo,
+        s"$curPath.additionalInfo")
+
+      (diffs flatten) ::: additionalInfoDiff
+    }
+
+    /**
+      * Compare additional information from ControlMeasureMetadata
+      * @param was     Value it had in reference
+      * @param is Value it has now
+      * @param curPath Path to the AdditionalInfo through the model
+      * @return Returns a list of model differences
+      */
+    private def additionalInfoComparison(was: Map[String, String],
+                                         is: Map[String,String],
+                                         curPath: String): List[ModelDifference[_]] = {
+      was.flatMap {
+        case (wasKey, wasValue) if wasKey == "conform_enceladus_version" =>
+          logVersionAndContinue("Conformance", wasValue, is.get(wasKey))
+        case (wasKey, wasValue) if wasKey == "std_enceladus_version" =>
+          logVersionAndContinue("Standartization", wasValue, is.get(wasKey))
+        case (wasKey, _) if wasKey == "std_application_id" || wasKey == "conform_application_id" => None
+        case (wasKey, wasValue) =>
+          is.get(wasKey) match {
+            case Some(isValue) if wasValue != isValue => Some(ModelDifference(s"$curPath.$wasKey", wasValue, isValue))
+            case None                                 => Some(ModelDifference(s"$curPath.$wasKey", wasValue, "Null"))
+            case _                                    => None
+          }
+      }.toList
+    }
+
+    private def logVersionAndContinue(name: String, refVersion: String, newVersion: Option[String]): Option[Nothing] = {
+      logger.info(s"$name versions is:")
+      logger.info(s"Reference - $refVersion")
+      logger.info(s"New - $newVersion")
+      None
+    }
+  }
+
+  /**
+    * Checkpoint's class extension adding compareWith
+    * @param checkpoint Checkpoint instance
+    */
+  implicit class CheckpointOps (checkpoint: Checkpoint){
+    /**
+      * Compare this Checkpoint with the one passed in
+      * @param secondCheckpoint Second checkpoint
```

Comment:
I'd call it `otherCheckpoint`

Reply:
so `otherCheckpoint`, `otherMeasurement`, etc.?